File: test-suite/src/test/java/io/micronaut/security/docs/securityrule/BodySecured.java
package io.micronaut.security.docs.securityrule;

//tag::clazz[]
import io.micronaut.aop.Around;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Around
@Documented
public @interface BodySecured {
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/securityrule/BodySecured.kt
package io.micronaut.security.docs.securityrule

import io.micronaut.aop.Around

//tag::clazz[]
@Target(AnnotationTarget.FUNCTION)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Around
@MustBeDocumented
annotation class BodySecured
//end::clazz[]